### PR TITLE
Prepare StaticAssetsContentSourceVc for basePath

### PR DESCRIPTION
**Pull request by jridgewell**
### Description

This is part 1 of WEB-993 `basePath` support. The static assets (things served in the `/public` directory) can only serve requests that include the `basePath`, so we'll need to pass that in. In order to not completely invalidate the [source](https://github.com/vercel/next.js/blob/2b1f0d9351610b04d01638efed19252ca81d0023/packages/next-swc/crates/next-dev/src/lib.rs#L379-L380) function (and all the things that depend on cells created there, ie everything) every time the user edits the `next.config.js`, we'll need to accept a `StringVc`. This sets us up without breaking backwards compat, making this multi-PR adventure a little easier.

### CODE REVIEW

File: crates/turbopack-dev-server/src/source/static_assets.rs [hunk 1]
```diff
@@ -16,19 +16,27 @@ use super::{
...
 pub struct StaticAssetsContentSource {
-    pub prefix: String,
+    pub prefix: StringVc,
     pub dir: FileSystemPathVc,
 }
 
 #[turbo_tasks::value_impl]
 impl StaticAssetsContentSourceVc {
     #[turbo_tasks::function]
     pub fn new(prefix: String, dir: FileSystemPathVc) -> StaticAssetsContentSourceVc {
```
Comment (ForsakenHarmony) [inline crates/turbopack-dev-server/src/source/static_assets.rs:26]: Is the plan to remove this later?
The different here only being `String` vs `StringVc` make the naming confusing

File: crates/turbopack-dev-server/src/source/static_assets.rs [hunk 2]
```diff
@@ -71,24 +80,29 @@ impl Introspectable for StaticAssetsContentSource {
...
-                            *path,
-                        )
-                        .into(),
-                        DirectoryEntry::Other(_) => todo!("what's DirectoryContent::Other?"),
-                        DirectoryEntry::Error => todo!(),
-                    };
-                    (StringVc::cell(name.clone()), child)
-                })
-                .collect(),
+            DirectoryContent::Entries(entries) => {
```
Comment (ForsakenHarmony) [inline crates/turbopack-dev-server/src/source/static_assets.rs:83]: can we use an early return for `NotFound` to reduce the indent here?

**Review by ForsakenHarmony — APPROVED** [review]

File: crates/turbopack-dev-server/src/source/static_assets.rs (hunk 2 above)
Comment (jridgewell) [reply crates/turbopack-dev-server/src/source/static_assets.rs:83]: Done.

File: crates/turbopack-dev-server/src/source/static_assets.rs (hunk 1 above)
Comment (jridgewell) [reply crates/turbopack-dev-server/src/source/static_assets.rs:26]: Yup.

**Pull request merged**